GVFS/GVFS/CommandLine/DehydrateVerb.cs

```diff
@@ -145,31 +164,139 @@ protected override void Execute(GVFSEnlistment enlistment)
                 // Local cache and objects paths are required for TryDownloadGitObjects
                 this.InitializeLocalCacheAndObjectsPaths(tracer, enlistment, retryConfig, serverGVFSConfig: null, cacheServer: null);
 
-                if (this.TryBackupFiles(tracer, enlistment, backupRoot))
+                if (string.IsNullOrEmpty(this.Folders))
                 {
-                    if (this.TryDownloadGitObjects(tracer, enlistment, retryConfig) &&
-                        this.TryRecreateIndex(tracer, enlistment))
+                    this.RunFullDehydrate(tracer, enlistment, backupRoot, retryConfig);
+                }
+                else
+                {
+                    string[] folders = this.Folders.Split(new[] { ";" }, StringSplitOptions.RemoveEmptyEntries);
+
+                    if (folders.Length > 0)
                     {
-                        // Converting the src folder to partial must be the final step before mount
-                        this.PrepareSrcFolder(tracer, enlistment);
-                        this.Mount(tracer);
+                        if (cleanStatus)
+                        {
+                            this.DehydrateFolders(tracer, enlistment, backupRoot, folders);
+                        }
+                        else
+                        {
+                            this.ReportErrorAndExit("Must have a clean git status to dehydrate folders.");
+                        }
+                    }
+                    else
+                    {
+                        this.ReportErrorAndExit("No folders to dehydrate.");
+                    }
+                }
+            }
+        }
+
+        private void DehydrateFolders(JsonTracer tracer, GVFSEnlistment enlistment, string backupRoot, string[] folders)
+        {
+            List<string> foldersToDehydrate = new List<string>();
+
+            if (!this.ShowStatusWhileRunning(
+                () =>
+                {
+                    string ioError;
+                    foreach (string folder in folders)

```

Comment:
Do we need any checks in here for nested folders?

What happens if a user passes in `Foo\bar;Foo`?  Would the copy of `Foo\bar` first cause the copy of its parent to fail?

Reply:
Wrote a test for this case and it works without issue because after copying, the `Foo\bar` folder is deleted and will not be copied again when copying `Foo`.